src/components/credit-card-number/code/views/creditCardNumberView.jsx

```diff
@@ -0,0 +1,78 @@
+var React = require('react');
+var Payment = require('payment');
+
+module.exports = React.createClass({
+
+  intent: null,
+  formatting: false,
+
+  propTypes: {
+    accepted: React.PropTypes.array,
+    cardNumber: React.PropTypes.string,
+    cardType: React.PropTypes.oneOf(['amex', 'discover', 'mastercard', 'visa']),
+    defaultValue: React.PropTypes.string,
+    id: React.PropTypes.string,
+    label: React.PropTypes.string,
+    name: React.PropTypes.string,
+    errorMessage: React.PropTypes.string
+  },
+
+  getInitialState: function() {
+
+    var isValid = Payment.fns.validateCardNumber(this.props.defaultValue);
+    var cardType = Payment.fns.cardType(this.props.defaultValue);
+
+    return {
+      valid: isValid,
+      error: (isValid) ? null : this.props.errorMessage,
+      cardType: cardType
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      accepted: ['amex', 'discover', 'mastercard', 'visa'],
+      formatInput: false,
+      id: 'credit-card-number',
+      name: 'credit-card-number',
+      placeHolder: '• • • •   • • • •   • • • •   • • • •',
+      showImages: false,
+      errorMessage: 'Invalid Card Number'
+    };
```

Comment:
no default is supplied for defaultValue and not providing one on the element declaration displays the invalid error for some reason.